### PR TITLE
feat: add etcd_debugging_lease_total gauge metric

### PR DESCRIPTION
**Pull request by jimmy-bro**
The feature was discussed in #17874

### CODE REVIEW

File: server/lease/metrics.go [hunk 1]
```diff
@@ -49,11 +49,19 @@ var (
 			// 1 second -> 3 months
 			Buckets: prometheus.ExponentialBuckets(1, 2, 24),
 		})
+
+	leaseTotal = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "etcd_debugging",
+		Subsystem: "lease",
+		Name:      "total",
```
Comment (jimmy-bro) [inline server/lease/metrics.go:56]: I‘m wondering if `active_total`  would be a more suitable name for this metric. 
Cause i feel like this description is more precise.

**Comment by jimmy-bro** [comment]
Should I write unit tests? I'm willing to write unit tests for all the metrics in https://github.com/etcd-io/etcd/blob/main/server/lease/metrics.go

File: server/lease/lessor.go [hunk 2]
```diff
@@ -303,6 +303,7 @@ func (le *lessor) Grant(id LeaseID, ttl int64) (*Lease, error) {
 
 	leaseTotalTTLs.Observe(float64(l.ttl))
 	leaseGranted.Inc()
+	leaseTotal.Inc()
```
Comment (gojoy) [inline server/lease/lessor.go:306]: should leaseTotal Inc when etcdserver restart ? the lease may recover from backend in [initAndRecover](https://github.com/etcd-io/etcd/blob/main/server/lease/lessor.go#L790)

Comment (jimmy-bro) [reply server/lease/lessor.go:306]: Thanks for your mentioning @gojoy ,I already updated the commit .
When it recover from backend, the metric will reset the value through leaseTotal.Set(float64(len(le.leaseMap))),
I used to considered doing leaseTotal.Inc inside the recover for-loop, but that would likely have a significant impact.

Comment (gojoy) [reply server/lease/lessor.go:306]: LGTM

**Comment by jmhbnz** [comment]
> Should I write unit tests? I'm willing to write unit tests for all the metrics in https://github.com/etcd-io/etcd/blob/main/server/lease/metrics.go

Yes please include tests, refer to https://github.com/etcd-io/etcd/pull/17983 for examples of unit and integration metrics tests.

**Pull request closed without merging**